src/App.tsx

```diff
@@ -59,40 +92,120 @@ function App() {
     }));
 
     setSortSettings(newSettings);
-    const filterByField = [...films].sort(
-      (a: any, b: any) => a[field] - b[field]
-    );
+    const filterByField = sortBy(films as any[], [], field as string);
     setFilterdFilms(filterByField.slice(0, 1));
     setIsShowFilter(!isShowFilter);
     setIsShowNextFilmBtn(true);
   };
 
   const onClickNextFilm = () => {
-    const field = sortSettings.reduce((acc, { isActive, field }) => {
-      return isActive ? field : acc;
-    }, "");
-    setFilterdFilms(
-      [...films]
-        .sort((a: any, b: any) => a[field] - b[field])
-        .slice(0, filterdFilms.length + 1)
-    );
+    const field = sortSettings.find((el) => el.isActive)?.field;
+    setFilterdFilms(sortBy(films, filterdFilms, field));
   };
 
   const handlerSearchFilter = (text: string) => {
     setInputValueFilter(text);
   };
+
+  const onChangeHandlerSelectCountre = (valueCountry: string) => {
+    setSelectValueCountry(valueCountry);
+  };
+
+  const onChangeHandlerFromToYear = (
+    value: string,
+    type: string,
+    field: string
+  ) => {
+    console.log(field);
+
+    const newSortFromToYear = {
+      ...sortFromToYear,
+      from: type === "from" ? value : sortFromToYear.from, //можно сделать от минимального года до сегодняшней даты
+      to: type === "to" ? value : sortFromToYear.to, //можно сделать от минимального года до сегодняшней даты
+    };
+    setSortFromToYear(newSortFromToYear);
+  };
+  const onChangeHandlerFromToRating = (
+    value: string,
+    type: string,
+    field: string
+  ) => {
+    console.log(field);
+
+    const newSortFromToRating = {
+      ...sortFromToRating,
+      from: type === "from" ? value : sortFromToRating.from, //можно сделать от минимального года до сегодняшней даты
+      to: type === "to" ? value : sortFromToRating.to, //можно сделать от минимального года до сегодняшней даты
+    };
+    setSortFromToRating(newSortFromToRating);
+  };
+  const onChangeHandlerGenre = (value: string) => {
+    setGenreList([...genreList, value]);
+  };
+  const onClickDeleteGenre = (value: string) => {
+    const index = genreList.indexOf(value);
+    const newList = [...genreList];
+    newList.splice(index, 1);
```

Comment:
are you sure this expression (newList) is already with a slice applied?

slice return new array